mlir/lib/Catalyst/Transforms/QnodeToAsyncPatterns.cpp

```diff
@@ -31,6 +31,135 @@ namespace catalyst {
 struct CallOpToAsyncOPRewritePattern : public mlir::OpRewritePattern<func::CallOp> {
     using mlir::OpRewritePattern<func::CallOp>::OpRewritePattern;
 
+    void insertDropRefOps(func::CallOp op, async::ExecuteOp executeOp,
+                          PatternRewriter &rewriter) const
+    {
+        PatternRewriter::InsertionGuard insertGuard(rewriter);
+        // Let's first place the awaits just before the terminator of this block.
+        // If there is no terminator in this block, then we will place the await
+        // at the end of the block.
+        Block *block = op->getBlock();
+        // TODO: Once we update LLVM versions, we might have access to function
+        // `mightHaveTerminator` For now, we have ''inlined'' the definition here.
+        bool mightHaveTerminator =
+            !block->empty() && block->back().mightHaveTrait<OpTrait::IsTerminator>();
+        if (mightHaveTerminator) {
+            rewriter.setInsertionPoint(block->getTerminator());
+        }
+        else {
+            rewriter.setInsertionPointToEnd(block);
+        }
+
+        // TODO: Here we place an await until the very end.
+        // The reason for this is because if the call returns no values,
+        // then we will never execute it.
+        // Potentially we could just remove the call altogether.
+        // But we should make sure that there are no side effects in the call.
+        // One possible side effect is printing to stdout.
```

Comment:
```suggestion
        // Here we place an await until the very end.
        // The reason for this is because if the call returns no values,
        // then we will never execute it.
        // TODO: We could just remove the call altogether.
        // But we should make sure that there are no side effects in the call.
        // One possible side effect is printing to stdout.
```